test/integration/instrumentation_test.rb

```diff
@@ -37,6 +37,25 @@ class InstrumentationTest < ActiveSupport::TestCase
     assert_event events.first, "claim", process_id: process.id, job_ids: jobs.map(&:id), claimed_job_ids: jobs.map(&:id), size: 3
   end
 
+  test "polling emits events" do
+    3.times { StoreResultJob.perform_later(42) }
+    process = nil
+    jobs = SolidQueue::Job.last(3)
+
+    events = subscribed("polling.solid_queue") do
+      worker = SolidQueue::Worker.new.tap(&:start)
+
+      wait_while_with_timeout!(3.seconds) { SolidQueue::ReadyExecution.any? }
+      process = SolidQueue::Process.last
+
+      worker.stop
+      wait_for_registered_processes(0, timeout: 3.second)
+    end
+
+    assert_equal 5, events.size
+    assert_event events.last, "polling"
```

Comment:
Nice! I think you can remove a few variables here, like `process`, `jobs`, `worker`, to make this test simpler, since we don't need them to check anything.

Reply:
ok! I removed as many as I could to get a minimum running example. I think `process` still needs to be there!